mock_ros2_project/action_client.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+ROS2 Action Client Example
+
+This node demonstrates how to create an action client that calls
+the Fibonacci action and handles feedback.
+"""
+
+import sys
+from typing import Optional
+
+import rclpy
+from rclpy.action import ActionClient
+from rclpy.node import Node
+from example_interfaces.action import Fibonacci
+
+
+class ActionClientNode(Node):
+    """Action client node that requests Fibonacci sequences."""
+    
+    def __init__(self) -> None:
+        """Initialize the action client node."""
+        super().__init__('action_client')
+        
+        # Create action client for Fibonacci action
+        self._action_client = ActionClient(self, Fibonacci, 'fibonacci')
+
+    def send_goal(self, order: int) -> None:
+        """Send a goal to the action server.
+        
+        Args:
+            order: The order of the Fibonacci sequence to generate
+        """
+        goal_msg = Fibonacci.Goal()
+        goal_msg.order = order
+        
+        if not self._action_client.wait_for_server(timeout_sec=5.0):
+            self.get_logger().error('Action server not available after waiting')
+            return
+        
+        self.get_logger().info(f'Sending goal: order={order}')
+        
+        self._send_goal_future = self._action_client.send_goal_async(
+            goal_msg,
+            feedback_callback=self.feedback_callback
+        )
+        
+        self._send_goal_future.add_done_callback(self.goal_response_callback)
+
+    def goal_response_callback(self, future) -> None:
+        """Handle goal response from action server.
+        
+        Args:
+            future: Future object containing the goal response
+        """
+        goal_handle = future.result()
+        
+        if not goal_handle.accepted:
+            self.get_logger().info('Goal rejected :(')
+            return
+        
+        self.get_logger().info('Goal accepted :)')
+        
+        self._get_result_future = goal_handle.get_result_async()
+        self._get_result_future.add_done_callback(self.get_result_callback)
+
+    def get_result_callback(self, future) -> None:
+        """Handle result from action server.
+        
+        Args:
+            future: Future object containing the action result
+        """
+        result = future.result().result
+        self.get_logger().info(f'Result: {result.sequence}')
+        rclpy.shutdown()
```

Comment:
Calling `rclpy.shutdown()` inside the result callback works, but keep in mind that the `finally` block later calls `rclpy.try_shutdown()` as well.  You may want to centralise shutdown logic in one place (e.g. only in the `finally` block) to avoid duplicated teardown or unexpected behaviour in more complex nodes.

```suggestion
        result = future.result().result
        self.get_logger().info(f'Result: {result.sequence}')
```